cellengine/resources/experiment.py

```diff
@@ -210,27 +211,50 @@ def undelete(self):
 
     @property
     def active_compensation(self) -> Optional[Union[Compensation, int]]:
+        """The most recently used compensation.
+
+        Accepted (case-insensitive) values are one of:
+        [
+            A Compensation object,
+            A Compensation ID,
+            'UNCOMPENSATED',
+            0
+            'FILE_INTERNAL',
+            -1
+            'PER_FILE',
+            -2
+        ]
+
+        """
         active_comp = self._active_comp
-        if type(active_comp) is str:
-            return ce.APIClient().get_compensation(self._id, active_comp)
-        elif type(active_comp) is int:
+        if isinstance(active_comp, (Compensation, int)):
             return active_comp
+        elif isinstance(active_comp, str):
+            return ce.APIClient().get_compensation(self._id, active_comp)
         elif active_comp is None:
             return None
-        raise ValueError(
-            f"Value '{active_comp}' is not a valid for activeCompensation."
-        )
+        else:
+            raise ValueError(
+                f"Value '{active_comp}' is not valid for active_compensation."
+            )
 
     @active_compensation.setter
-    def active_compensation(self, compensation: Union[Compensation, int]):
-        if type(compensation) is Compensation:
-            self._active_comp = compensation._id
-        elif (
-            compensation == "UNCOMPENSATED"
-            or compensation == "FILE_INTERNAL"
-            or compensation == "PER_FILE"
-        ):
+    def active_compensation(self, compensation: Union[Compensation, int, str]):
+        """Set the active compensation."""
+        special_values = {"uncompensated": 0, "file_internal": -1, "per_file": -2}
+        if isinstance(compensation, Compensation):
+            self._active_comp = compensation
```

Comment:
Untested: does this work correctly (does it send the Compensation object's `_id`) if you now save the experiment back to the server?

This (here and line 232-233R) is the first case that I recall where a relationship property is morphed into the related object. I don't think we do that with `gate.experimentId` or `population.parentId` or anything else, right?

Reply:
I think so... rule of least surprising behavior/consistency.